src/index.js

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import { Menu } from "./components/Menu";
+import { OrderSummary } from "./components/Order";
+import "./styles/main.scss";
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      orderItems: [],
+      orderTotal: 0.0
+    };
+    this.addToOrder = this.addToOrder.bind(this);
+    this.removeFromOrder = this.removeFromOrder.bind(this);
+  }
+
+  addToOrder(name, price, bread, isHot) {
+    const orderItems = this.state.orderItems;
+    const orderTotal = this.state.orderTotal;
+
+    orderItems.push({
+      name: name,
+      price: price,
+      bread: bread,
+      isHot: isHot
+    });
+
+    this.setState({
+      orderItems: orderItems,
```

Comment:
```js
this.setState(prevState => ({
  orderItems: [...prevState.orderItems, {newItem}]
}))
```

☝️ that might be a safer syntax when mutating the state on an array

Reply:
Is `prevState` a fancy React function? this makes sense though man

Reply:
Yeah, just helps to protect when mutating state. Its just an argument containing the previous current state that gets passed to the `setState` function

Reply:
what do the 3 dots do here man?, I've seen this before somewhere… `[...prevState.orderItems`
Would the same apply to the `RemoveFromOrder` method?

Reply:
Spreads the current array, into a new array with your new added item 👍 The remove function would be similar, maybe using the `.filter` function on `prevState`?